packages/engine-server/src/metadata/service.ts

```diff
@@ -98,6 +122,21 @@ export class MetadataService {
     return false;
   }
 
+  getGlobalVersion() {
+    return this.getMeta().version || "0.0.0";
+  }
+
+  getLapsedUserSurveyStatus() {
+    return this.getMeta().lapsedUserSurveyStatus;
+  }
+
+  getActivationContext() {
+    return (
+      this.getMeta().workspaceActivationContext ??
+      WorkspaceActivationContext.normal
+    );
+  }
+
   setMeta(key: keyof Metadata, value: any) {
```

Comment:
Just a Q:
How can we add metadata if the key is dynamic? 
eg:  to store hierarchy destination for imported google doc. (key: note.fname, value: user input)

Reply:
we can't (or if we do, it would no longer be typed). for this use case, would have the key be "pods.gdoc" and the value be an object that can have dynamic keys

Reply:
something like

```js
type Metadata {
  ...
  "pods.gdoc": Record<string, string>
}
```

or

```js
type PodsGDoc = {[key: string]: string};
type Metadata {
  ...
  "pods.gdoc": PodsGDoc
  ...
}
```

not entirely sure of the implications of using a `Record` type vs a regular index signature, but both seem identical on the surface.

Reply:
Got it. Thank you